python/ray/autoscaler/_private/_azure/config.py

```diff
@@ -218,22 +223,105 @@ def _configure_resource_group(config):
 
 
 def _configure_key_pair(config):
+    """
+    Configure SSH keypair, generate one automatically if keypair is not specified and does not exist.
+    """
     ssh_user = config["auth"]["ssh_user"]
     public_key = None
-    # search if the keys exist
-    for key_type in ["ssh_private_key", "ssh_public_key"]:
-        try:
-            key_path = Path(config["auth"][key_type]).expanduser()
-        except KeyError:
-            raise Exception("Config must define {}".format(key_type))
-        except TypeError:
-            raise Exception("Invalid config value for {}".format(key_type))
 
-        assert key_path.is_file(), "Could not find ssh key: {}".format(key_path)
+    # Check if user specified custom SSH key paths
+    user_specified_private_key = "ssh_private_key" in config["auth"]
+    user_specified_public_key = "ssh_public_key" in config["auth"]
+
+    # Validate that the user either specfied both keys or none, but not just one
```

Comment:
A bonus bit of foo could be to auto-get the user from the OS environment if it's not set.

I couldn't get it to work! I got this far:

```
from getpass import getuser
...
    if not user_specified_ssh_user:
        # If not specified, use the current system user as the default SSH user
        ssh_user = getuser()
        config["auth"] = config.get("auth", {})
        config["auth"]["ssh_user"] = str(ssh_user)
```